distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date_Time.enso

```diff
@@ -326,6 +328,24 @@ type Date_Time
     day_of_week self =
         Day_Of_Week.from (Time_Utils.get_field_as_zoneddatetime self ChronoField.DAY_OF_WEEK) Day_Of_Week.Monday
 
+    ## Returns the first date within the `Time_Period` or `Date_Period`
+       containing self.
+    start_of : (Date_Period|Time_Period) -> Date_Time
+    start_of self period=Date_Period.Month =
+        adjusted = period.adjust_start self
+        case period.is_date_period of
```

Comment:
Can we use `is_a Date_Period`?

Reply:
![Screenshot from 2022-09-10 14-00-32](https://user-images.githubusercontent.com/1436948/189490646-3023ee00-c689-41d0-9f36-4b840b2d2e25.png)

Unfortunately not. Until we rework `is_a` to work with types.